### PR TITLE
Add feature of not changing color scheme.

### LINKED ISSUE #297: Feature Request - Not changing the view's colour scheme
If color_scheme is set to null, PlainTasks should not change the colour scheme of the view.

### PR DESCRIPTION
**Pull request by pradyunsg**
If color_scheme is set to null, it is not changed.

Resolves #297

### CODE REVIEW

**Comment by vovkkk** [comment]
What practical issue you’re trying to resolve?

I’m reluctant to merge it because first of all this scheme change is applied to created document only, so it could cause confusing issue:
1. user creates new doc (it has normal scheme)
2. then somehow set scheme in settings to null (scheme of doc remains)
3. open existing todo file which is not highlighted as (1)

It is edge case, but confusing. Currently code mimic default behaviour and it seems right.

**Comment by pradyunsg** [comment]
That I have a made my own colour schemes and they support PlainTasks. I use
a shortcut to cycle between them. Now, when I change the color scheme (in
Preferences) I want all views to change color, which they mostly do. Since
PlainTasks has a separate setting for color scheme, I would also have to
write a plugin that modifies that setting along with the Preferences one...

This feature would allow me to get away without any changes to my current
structure. Plus it shouldn't break anyone else'e setup. Essentially,
setting color_scheme to null means inherit from preferences (i.e. Don't set
on view)

I'm not too sure if the PR code works exactly as I think it should...

**Comment by vovkkk** [comment]
> setting color_scheme to null means inherit from preferences

No, it does not, at least on Windows setting scheme to null results in black text on white background.

I’m closing this pr, because it may cause confusion and, indeed, it does not work the way you wanted.

Unfortunately, ST have no way to find where exactly certain setting is stored, thus you have to add special case for PT in your plugin, something like:

```python
sublime.load_settings('PlainTasks.sublime-settings').set('color_scheme', 'path/to/scheme')
sublime.save_settings('PlainTasks.sublime-settings')
```

**Comment by pradyunsg** [comment]
Just don't set color_scheme on the plain tasks view if color_scheme is null
in the plain tasks preferences. Then, the view will, automatically, use the
one in the global Preferences.

I guess I was not very clear in that previous comment.

-via phone

On Sun, Nov 13, 2016, 12:56 Vova Kolobok <notifications@github.com> wrote:

Closed #298 <https://github.com/aziz/PlainTasks/pull/298>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/aziz/PlainTasks/pull/298#event-856944506>, or mute the
thread

**Comment by pradyunsg** [comment]
If I want to edit it, I'll change this... 

setting color_scheme to null means inherit from preferences

To

setting color_scheme in the plain tasks preferences to null would then mean that plain tasks won't touch the color scheme.

**Comment by vovkkk** [comment]
I have a feeling that we do not understand each other.
Could you show a code of the mentioned plugin?

**Comment by pradyunsg** [comment]
>  Could you show a code of the mentioned plugin?

It's https://github.com/damccull/sublimetext-SolarizedToggle . Not my plugin, which is why I don't want to be writing a plugin just for this package.

> I have a feeling that we do not understand each other.

Me too. I've move to the laptop. So, let me try again...

---

I have an need for some way of telling PlainTasks to not use a different colour scheme for the Tasks views and use the one set in `Preferences.sublime-settings`, without having to specify the same scheme (the path) in `PlainTask.sublime-settings`.

I frequently toggle between 2 colour schemes, both of which I have made. I have added support for the Tasks syntax in them. Since I use the aforementioned plugin, I cannot change the settings of PlainTasks unless I modify that plugin, which would be out of scope so, I'll have to create and maintain my own version of it.

So, I thought if PlainTasks allowed me to signal "don't fiddle with colour schemes, just let it be", it would be a neater solution to this issue.

Does that make the situation clearer?

**Comment by vovkkk** [comment]
> way of telling PlainTasks to not use a different colour scheme for the Tasks views and use the one set in Preferences.sublime-settings, without having to specify the same scheme (the path) in PlainTask.sublime-settings.

It is possible only if `PlainTask.sublime-settings` file(s) do not exist or do not contain `"color_scheme"` setting.
Note that absence of color_scheme and `"color_scheme": null` are not the same thing.

We cannot remove color_scheme from default settings file, because we want plugin work out-of-box for average user without any steps to setup.

You can write separate command in `User/MySolarizedToggle.py` something like this
```python
import sublime
import sublime_plugin


class MySolarizedToggleCommand(sublime_plugin.ApplicationCommand):
    def run(self):
        sublime.run_command('solarized_toggle')
        sublime.load_settings('PlainTasks.sublime-settings').set('color_scheme', sublime.load_settings('Preferences.sublime-settings').get('color_scheme'))
        sublime.save_settings('PlainTasks.sublime-settings')
```
and in user keybinding file
```js
[
	{
		"keys": ["f12"], "command": "my_solarized_toggle"
	}
]
```

**Comment by pradyunsg** [comment]
You could special case color_scheme being null... Which is what I've been
trying to say...

And, I didn't really think about reusing through the api... Huh. That'll
also do just fine.

**Comment by pradyunsg** [comment]
Thanks for the help!

**Comment by vovkkk** [comment]
> You could special case color_scheme being null... Which is what I've been trying to say...

And I’ve been trying to say that it could lead to confusion (see my first message), because when you do `settings.get()` then absence of setting and null of its value are the same thing but from user pov they are not the same because setting scheme to null results in black text on white background.

**Comment by pradyunsg** [comment]
And, you think someone might set it to null?

**Comment by vovkkk** [comment]
Yes, I could imagine some weird plugin which go thru all settings files and set value which for some reason/mistake was evaluated to None.

**Comment by pradyunsg** [comment]
Well, no comments on how I find that argument. I still think the sane thing to do here would be to use the scheme from preferences...

**Comment by vovkkk** [comment]
If we set scheme from preferences for newly created doc and then user open existing todo file, then user will see two todo tabs with different colours. If it is sane then I’m sorry.

**Comment by pradyunsg** [comment]
> Note that absence of color_scheme and "color_scheme": null are not the same thing.

"absence of color_scheme" would never happen if it's set in the default settings file. So, if it's `None` at the API level, we can be sure it's because the user set it as null, explicitly in the file. I did not communicate this. I don't think you missed this but anyway... :stuck_out_tongue_winking_eye:

Anyway, my setup works fine for me. Just going around again.
